src/lib/group.ts

```diff
@@ -167,15 +270,10 @@ class Group {
    * ```
    */
   equals(g: Group) {
-    if (this.#isConstant() && g.#isConstant()) {
-      let { x: x1, y: y1 } = this;
-      let { x: x2, y: y2 } = g;
+    let { x: x1, y: y1 } = this;
+    let { x: x2, y: y2 } = g;
 
-      return Bool(x1.equals(x2).and(y1.equals(y2)));
-    } else {
-      let z = Snarky.group.equals(this.#toTuple(), g.#toTuple());
-      return Bool.Unsafe.ofField(new Field(z));
-    }
+    return x1.equals(x2).and(y1.equals(y2));
```

Comment:
if we already break the constraints, might as well move this part to JS entirely